core/src/main/java/com/linecorp/armeria/client/retry/RetryingClientBuilder.java

```diff
@@ -75,35 +86,54 @@ public RetryingClientBuilder useRetryAfter(boolean useRetryAfter) {
      * Sets the length of content required to determine whether to retry or not. If the total length of content
      * exceeds this length and there's no retry condition matched, it will hand over the stream to the client.
      * Note that this property is useful only if you specified {@link RetryRuleWithContent} when calling
-     * this builder's constructor. The default value of this property is
-     * {@value #DEFAULT_CONTENT_PREVIEW_LENGTH}.
+     * this builder's constructor. The default value of this property is {@value #DEFAULT_MAX_CONTENT_LENGTH}.
      *
-     * @param contentPreviewLength the content length to preview. {@code 0} does not disable the length limit.
+     * @throws IllegalStateException if this builder is created with a {@link RetryRule} rather than
+     *                               {@link RetryRuleWithContent}
+     * @throws IllegalArgumentException if the specified {@code maxContentLength} is equal to or
+     *                                  less than {@code 0}
      *
-     * @return {@link RetryingClientBuilder} to support method chaining
+     * @deprecated Use {@link RetryingClient#builder(RetryRuleWithContent, int)}.
+     */
+    @Deprecated
+    public RetryingClientBuilder maxContentLength(int maxContentLength) {
+        checkState(!isMaxContentLengthSet, "maxContentLength is already set by " +
+                                           "RetryingClient.builder(RetryRuleWithContent, maxContentLength)");
+        checkState(needsContentInRule, "cannot set maxContentLength when RetryRule is used; " +
+                                       "Use RetryRuleWithContent to enable this feature.");
+        checkArgument(maxContentLength > 0,
+                      "maxContentLength: %s (expected: > 0)", maxContentLength);
+        this.maxContentLength = maxContentLength;
+        return this;
+    }
+
+    /**
+     * Sets the length of content required to determine whether to retry or not. If the total length of content
+     * exceeds this length and there's no retry condition matched, it will hand over the stream to the client.
+     * Note that this property is useful only if you specified {@link RetryRuleWithContent} when calling
+     * this builder's constructor. The default value of this property is {@value #DEFAULT_MAX_CONTENT_LENGTH}.
+     *
+     * @param contentPreviewLength the content length to preview. {@code 0} does not disable the length limit.
      *
      * @throws IllegalStateException if this builder is created with a {@link RetryRule} rather than
      *                               {@link RetryRuleWithContent}
-     * @throws IllegalArgumentException if the specified {@code contentPreviewLength} is equal to or
+     * @throws IllegalArgumentException if the specified {@code maxContentLength} is equal to or
      *                                  less than {@code 0}
+     *
+     * @deprecated Use {@link #maxContentLength(int)}.
```

Comment:
I think we should guide to use `builder(...)` not the deprecated one.

Reply:
Oops, I should remove this method.
```java
RetryClient.builder(RetryRuleWithContent); // Will use default max content length value
RetryClient.builder(RetryRuleWithContent, maxContentLength); // Will use the specified value

// Deprecated
RetryClient.builder(RetryRuleWithContent).contentPreviewLength(maxContentLength);
```